src/Components/CreateImageWizard/steps/msAzure.beta.js

```diff
@@ -0,0 +1,255 @@
+          </Text>
+          <Text>
+            To authorize Image Builder to push images to Microsoft Azure, the
+            account owner must configure Image Builder as an authorized
+            application for a specific tenant ID and give it the role of
+            &quot;Contributor&quot; to at least one resource group.
+            <br />
+            <Button
+              component="a"
+              target="_blank"
+              variant="link"
+              icon={<ExternalLinkAltIcon />}
+              iconPosition="right"
+              isInline
+              href="https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-auth-code-flow"
+            >
+              Learn more about OAuth 2.0
+            </Button>
+          </Text>
+        </TextContent>
+      ),
+    },
+    {
+      component: componentTypes.RADIO,
+      label: 'Share method:',
+      name: 'azure-type',
+      initialValue: 'azure-type-source',
+      autoFocus: true,
+      options: [
+        {
+          label: 'Use an account configured from Sources.',
+          description:
+            'Use a configured source to launch environments directly from the console.',
+          value: 'azure-type-source',
+          'data-testid': 'azure-radio-source',
+          autoFocus: true,
+        },
+        {
+          label: 'Manually enter the account information.',
+          value: 'azure-type-manual',
+          'data-testid': 'azure-radio-manual',
+          className: 'pf-u-mt-sm',
+        },
+      ],
+    },
+    {
+      component: 'azure-sources-select',
+      name: 'azure-sources-select',
+      className: 'pf-u-max-width',
+      label: 'Source Name',
+      isRequired: true,
+      validate: [
+        {
+          type: validatorTypes.REQUIRED,
+        },
+      ],
+      condition: {
+        when: 'azure-type',
+        is: 'azure-type-source',
+      },
+    },
+    {
+      component: componentTypes.PLAIN_TEXT,
+      name: 'azure-sources-select-description',
+      label: <SourcesButton />,
+      condition: {
+        when: 'azure-type',
+        is: 'azure-type-source',
+      },
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'azure-tenant-id',
+      className: 'pf-u-w-50',
+      'data-testid': 'azure-tenant-id',
+      type: 'text',
+      label: 'Azure Tenant GUID',
+      isRequired: true,
+      autoFocus: true,
+      validate: [
+        {
+          type: validatorTypes.REQUIRED,
+        },
+        {
+          type: validatorTypes.PATTERN,
+          pattern:
+            /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i,
+          message: 'Please enter a valid tenant ID',
+        },
+      ],
+      condition: {
+        when: 'azure-type',
+        is: 'azure-type-manual',
+      },
+    },
+    {
+      component: componentTypes.TEXT_FIELD,
+      name: 'azure-tenant-id',
+      className: 'pf-u-w-50',
+      'data-testid': 'azure-tenant-id',
```

Comment:
Would it make the tests a bit more robust/explicit if we gave these two fields different test ids? Like `azure-tenant-id-source` and `-manual`?

Reply:
yup, make sense, I've change the ids :+1: